### ISSUE TITLE
does not work for extra-dep of type git?

### ISSUE THREAD

**Issue by jwaldmann** [snoyberg/githash#18]
Hi. 

I am using githash in an "extra-dep" of type "git".
`stack install` for the main project fails while compiling the depency: 
```
...
cyp                              > /tmp/stack-6210c3ddc6e727eb/cyp-0.3.0/src/Test/Info2/Cyp/Module.hs:67:12: error:
cyp                              >     • Exception when trying to run compile-time code:
cyp                              >         GHEGitRunFailed "." ["rev-parse","--show-toplevel"] (ExitFailure 128) "" "fatal: not a git repository (or any parent up to mount point /)\nStopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).\n"
cyp                              >       Code: tGitInfoCwd     
...
```
Is there a way to make this work?

This is with ghc-8.6.5, lts-14.27.

**Comment by jwaldmann** [comment]
The equivalent program with `gitrev` seems to be working.

**Comment by snoyberg** [comment]
I don't think it works with `gitrev`, it's just not returning an error message. That was one of the limitations `githash` fixed.

**Comment by jwaldmann** [comment]
I made this test: https://github.com/jwaldmann/test-gitrevhash
(trivial project, it just pulls in the dependency that uses gitrev).

When I do `stack run` on a machine with stack-1.9.3 it does print
```
master@234a22bff0437e9e357a251125fcfb39da83b5ee (Tue Apr 14 21:13:28 2020 +0200)
```
This is correct. On a machine with stack-2.1.3.1 I get
```
UNKNOWN@UNKNOWN (UNKNOWN)
```
both using ghc-8.6.5.

**Comment by snoyberg** [comment]
This is expected behavior with Stack 2, which no longer builds in the context of a full Git repository.

**Comment by jwaldmann** [comment]
Thanks for the explanation. So what could be done about making githash work with stack-2? I still think it'd be a useful feature. If someone, perhaps me, finds the time to work on this, what approach do you recommend?

**Comment by snoyberg** [comment]
This is honestly not a use case I ever considered. I use `githash` in applications, not in libraries pulled in as extra-deps. When using the `git` behavior from Stack, the `.git` directory simply isn't available when building.

**Comment by jwaldmann** [comment]
I could probably print git info for the containing application, and from that, reconstruct git info of the external module. It is written in stack.yaml, after all.

On the other hand, ".git directory isn't available" - how hard would it be to get it, or just the part of it that's neeed? 

Could it be done when evaluationg the slices? Perhaps that's too expensive. 

Or could `stack` provide it, given some special option? Perhaps that's too much work for stack (not implemetation, as it's already been there, but maintenance).

**Comment by snoyberg** [comment]
It can't be handled in this library at all, this would be entirely part of Stack. It's theoretically possible you could set up some flag in Stack for this, but you're right that the maintenance burden is likely not worth it. One thing you could try is including a few files from `.git` in your `extra-source-files` in your cabal file.

**Comment by jwaldmann** [comment]
Thanks. Would it work if I change the extra-dep into a git submodule? I will try this.

**Comment by snoyberg** [comment]
That should work, yes
